### PR TITLE
8329521: Serial: Rename MarkSweep to SerialFullGC

### PR DESCRIPTION
**Pull request by lgxbslgx**
Hi all,

This patch renames the `MarkSweep` to `SerialFullGC` and fixes some comments related to `MarkSweep`.

The tests `make test-tier1_gc` passed locally. Thanks for taking the time to review.

Best Regards,
-- Guoxiong

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8329521](https://bugs.openjdk.org/browse/JDK-8329521): Serial: Rename MarkSweep to SerialFullGC (**Enhancement** - P4)

### Reviewers
 * [Albert Mingkun Yang](https://openjdk.org/census#ayang) (@albertnetymk - **Reviewer**) ⚠️ Review applies to [29615768](https://git.openjdk.org/jdk/pull/18619/files/29615768519919d9357db7301b667bc645377e72)
 * [Ivan Walulya](https://openjdk.org/census#iwalulya) (@walulyai - **Reviewer**) ⚠️ Review applies to [29615768](https://git.openjdk.org/jdk/pull/18619/files/29615768519919d9357db7301b667bc645377e72)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/18619/head:pull/18619` \
`$ git checkout pull/18619`

Update a local copy of the PR: \
`$ git checkout pull/18619` \
`$ git pull https://git.openjdk.org/jdk.git pull/18619/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 18619`

View PR using the GUI difftool: \
`$ git pr show -t 18619`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/18619.diff">https://git.openjdk.org/jdk/pull/18619.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/18619#issuecomment-2036810328)

### CODE REVIEW

**Review by albertnetymk — APPROVED** [review]

File: src/hotspot/share/gc/serial/tenuredGeneration.cpp [hunk 1]
```diff
@@ -24,7 +24,7 @@
 
 #include "precompiled.hpp"
 #include "gc/serial/cardTableRS.hpp"
-#include "gc/serial/markSweep.hpp"
+#include "gc/serial/serialFullGC.hpp"
 #include "gc/serial/serialBlockOffsetTable.inline.hpp"
```
Comment (walulyai) [inline src/hotspot/share/gc/serial/tenuredGeneration.cpp:28]: Ordering of includes

**Review by walulyai — APPROVED** [review]
LGTM!

Minor nit.

File: src/hotspot/share/gc/serial/tenuredGeneration.cpp (hunk 1 above)
Comment (lgxbslgx) [reply src/hotspot/share/gc/serial/tenuredGeneration.cpp:27]: Fixed.

**Comment by lgxbslgx** [comment]
@albertnetymk @walulyai Thanks for the reviews.

/integrate

**Pull request closed without merging**